app/src/main/kotlin/com/wire/android/WireApplication.kt

```diff
@@ -71,6 +74,9 @@ class WireApplication : BaseApp() {
     @Inject
     lateinit var globalDataStore: Lazy<GlobalDataStore>
 
+    @Inject
+    lateinit var userDataStoreProvider: Lazy<UserDataStoreProvider>
```

Comment:
injecting this here will not work when switching accounts, meaning the current user data store object will be injecting when starting the app, and never changed

Reply:
Alternatively you can try assisted injection to get the current user data store everytime

Reply:
We are not getting the `UserDataStore` directly but rather its provider to then later listen to active account and get its respective `UserDataStore`

Reply:
nvm this only the provider not the direct user data store object